ddtrace/internal/_encoding.pyx

```diff
@@ -703,59 +706,79 @@ cdef class MsgpackEncoderV04(MsgpackEncoderBase):
         cdef Py_ssize_t L
         cdef int ret
         cdef dict d
+        cdef list m
 
-        if PyDict_CheckExact(meta):
-            d = <dict> meta
-            L = len(d) + (dd_origin is not NULL) + (len(span_events) > 0)
-            if L > ITEM_LIMIT:
-                raise ValueError("dict is too large")
+        if not PyDict_CheckExact(meta):
+            raise TypeError("Unhandled meta type: %r" % type(meta))
 
-            ret = msgpack_pack_map(&self.pk, L)
-            if ret == 0:
-                for k, v in d.items():
-                    ret = pack_text(&self.pk, k)
-                    if ret != 0:
-                        break
-                    ret = pack_text(&self.pk, v)
-                    if ret != 0:
-                        break
-                if dd_origin is not NULL:
-                    ret = pack_bytes(&self.pk, _ORIGIN_KEY, _ORIGIN_KEY_LEN)
-                    if ret == 0:
-                        ret = pack_bytes(&self.pk, dd_origin, strlen(dd_origin))
-                    if ret != 0:
-                        return ret
-                if span_events:
-                    ret = pack_text(&self.pk, SPAN_EVENTS_KEY)
-                    if ret == 0:
-                        ret = pack_text(&self.pk, span_events)
-            return ret
+        d = <dict> meta
 
-        raise TypeError("Unhandled meta type: %r" % type(meta))
+        # Filter meta to only str/bytes values
+        m = []
+        for k, v in d.items():
+            if PyUnicode_Check(v) or PyBytesLike_Check(v):
+                m.append((k, v))
+            else:
+                log.warning("Meta key %r has non-string value %r, skipping", k, v)
+
+        L = len(m) + (dd_origin is not NULL) + (len(span_events) > 0)
+        if L > ITEM_LIMIT:
+            raise ValueError("dict is too large")
+
+        ret = msgpack_pack_map(&self.pk, L)
+        if ret == 0:
+            for k, v in m:
+                ret = pack_text(&self.pk, k)
+                if ret != 0:
+                    break
+                ret = pack_text(&self.pk, v)
+                if ret != 0:
+                    break
+            if dd_origin is not NULL:
+                ret = pack_bytes(&self.pk, _ORIGIN_KEY, _ORIGIN_KEY_LEN)
+                if ret == 0:
+                    ret = pack_bytes(&self.pk, dd_origin, strlen(dd_origin))
+                if ret != 0:
+                    return ret
```

Comment:
nit: this can be a break too. It would be nice to be consistent

Reply:
This return statement isn't in a loop so I don't think `break` makes sense as a replacement.